c10/util/typeid.h

```diff
@@ -126,7 +123,8 @@ struct TypeMetaData final {
   using Delete = void(void*);
 
   TypeMetaData() = delete;
-  constexpr TypeMetaData(
+
+  TypeMetaData(
```

Comment:
Why not constexpr anymore?

Reply:
See the second note in the PR description - adding the second constructor caused clang to start erroring as if the `constexpr` annotation had been silently dropped from both constructors. The client code didn't really rely on it so I just dropped them - but FYI there's no second constructor in #44965 so it's back 😁